src/sentry/snuba/issue_platform.py

```diff
@@ -0,0 +1,78 @@
+from datetime import timedelta
+from typing import Dict, List, Optional, Sequence
+
+from sentry.snuba import discover
+from sentry.utils.snuba import Dataset, SnubaTSResult
+
+
+def query(
+    selected_columns,
+    query,
+    params,
+    snuba_params=None,
+    equations=None,
+    orderby=None,
+    offset=None,
+    limit=50,
+    referrer=None,
+    auto_fields=False,
+    auto_aggregations=False,
+    use_aggregate_conditions=False,
+    allow_metric_aggregates=True,
+    conditions=None,
+    functions_acl=None,
+    dry_run=False,
+    transform_alias_to_input_format=False,
+    has_metrics: bool = True,
+    use_metrics_layer: bool = False,
+):
+    return discover.query(
```

Comment:
I know it increases code duplication 😬 but could we put all the dataset specific code here instead of in snuba.discover? ie. instead of adding a `dataset` parameter to `discover.query`, create a QueryBuilder object directly here?